File: BB.Core/Model/Order.cs
using System;
using System.Collections.Generic;

namespace BB.Core.Model
{
    public class Order
    {
        public Order()
        {
            OrderItems = new List<OrderItem>();
            StatusUpdates = new List<StatusUpdate>();
        }

        public Guid OrderId { get; set; }

        public long UserID { get; set; }

        public virtual User User { get; set; }

        public virtual ICollection<OrderItem> OrderItems { get; set; }

        public virtual ICollection<StatusUpdate> StatusUpdates { get; set; }
    }
}

File: BB.Web/app/Model/Orders/Order.ts
import u = require("../Users/User");
import User = u.User;
import oi = require("./OrderItem");
import OrderItem = oi.OrderItem;
import su = require("../Status/StatusUpdate");
import StatusUpdate = su.StatusUpdate;

export class Order
{
    public OrderId: string;
    public UserId: number;
    public User: User;
    public OrderItems: OrderItem[];
    public StatusUpdate: StatusUpdate[];

}